hw3maps/cmd/main.go

```diff
@@ -0,0 +1,76 @@
+/*
+ 3. У учеников старших классов прошел контрольный срез по нескольким предметам. Выведите данные в читаемом виде
+
+в таблицу вида
+_____________________________________
+Student name  | Grade | Object    |   Result
+____________________________________
+Ann			  |     9 | Math	  |  4
+Ann 		  |     9 | Biology   |  4
+...
+
+Вводные данные представлены в файле dz3.json
+*/
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+)
+
+func main() {
+
+	file, err := os.Open("../dz3.json")
+
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer file.Close()
+
+	decoder := json.NewDecoder(file)
+
+	data := Data{}
+
+	err = decoder.Decode(&data)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	students := make(map[int]Student)
+
+	for _, val := range data.ListOfStudents {
+		students[val.Id] = Student{Id: val.Id,
+			Name:  val.Name,
+			Grade: val.Grade}
+	}
```

Comment:
здесь можно не объявлять новый объект, а просто передать сразу студента
```suggestion
	for _, student := range data.ListOfStudents {
		students[student.Id] = student
```

Reply:
аналогично для предметов